docs/api/networksocket/securesocket.md

```diff
@@ -0,0 +1,166 @@
+## Secure Socket
+
+Mbed OS provides an interface for creating secure connections in the form of TLS stream. The `TLSSocket` class provides the ability to secure any stream-based socket connection, for example TCP stream. This allows you to use existing protocol libraries through a secure connections.
+
+`TLSSocket` inherits the `Socket` class, which allows any application that uses `Socket` to use `TLSSocket` instead. Secure socket both uses Socket interface as its transport layer and implements it. This makes it transport independent, and there is no direct dependency on the IP stack. For example, you can use the HTTP library and give `TLSSocket` to it to covert it to HTTPS.
+
+### API
+
+`TLSSocketWrapper` implements Mbed OS Socket API and extends it with functions that allow configuring security certificates. Please note that for most of the use cases, you are using these methods through `TLSSocket` class.
+
+### Usage example
+
+The TLSSocket API follows the Socket API, so it is straightforward to use after setting up:
+
+```
+TLSSocket *socket = new TLSSocket();
+socket->open(network)
+socket->set_root_ca_cert(certificate);
+
+socket->connect(HOST_NAME, PORT)
+
+socket->send(data, size);
+```
+
+Please note that internal TLS structures require over 1 kB of RAM, so you need to allocate each TLSSocket from the heap by using the `new` command, instead of using stack or statically allocating it.
+
+### Design
+
+Internally `TLSSocket` consists of two classes, `TLSSocketWrapper` and `TLSSocket`, as shown in the following diagram:
+
+![TLSSocket UML](tlssocket.png)
+
+The `TLSSocketWrapper` can use any `Socket` as its transport. `TLSSocket` is a helper that uses directly `TCPSocket` for its transport, so you can adopt existing TCP based applications to TLS.
+
+One use case of `TLSSocketWrapper` is that existing TCP socket can be upgraded to TLS, by wrapping it like this:
+
+```
+TCPSocket connection(net);
```

Comment:
The constructor taking in a NetworkStack as a parameter is now deprecated, can we modify this example to:
```
TCPSocket connection;
connection.open(net);
```
?

Reply:
Fixed, thanks for noticing.